Security_Engine/Create/CameraFieldOfView.cs

```diff
@@ -0,0 +1,364 @@
+
+        [Description("Returns the polycurve camera view cone.")]
+        [Input("cameraDevice", "The CameraDevice object to compute the camera view cone for.")]
+        [Output("conePolyCurve", "PolyCurve object that represents camera view cone.")]
+        public static PolyCurve CameraViewCone(this CameraDevice cameraDevice)
+        {
+            Point cameraLocation = cameraDevice.EyePosition;
+            Point targetLocation = cameraDevice.TargetPosition;
+            double radius = targetLocation.Distance(cameraLocation);
+            double horizontal = cameraDevice.HorizontalFieldOfView;
+            double angle = 2 * Math.Atan(horizontal / 2 / radius);
+
+            Vector direction = BH.Engine.Geometry.Create.Vector(cameraLocation, cameraDevice.TargetPosition);
+            Vector startPointDir = direction.Rotate(-angle / 2, Vector.ZAxis);
+            Vector endPointDir = direction.Rotate(angle / 2, Vector.ZAxis);
+            Point startPoint = cameraLocation + startPointDir;
+            Point endPoint = cameraLocation + endPointDir;
+
+            Arc coneArc = BH.Engine.Geometry.Create.Arc(startPoint, targetLocation, endPoint);
+            Line startLine = BH.Engine.Geometry.Create.Line(cameraLocation, startPoint);
+            Line endLine = BH.Engine.Geometry.Create.Line(endPoint, cameraLocation);
+
+            PolyCurve conePolyCurve = new PolyCurve();
+            conePolyCurve.Curves = new List<ICurve>() { startLine, coneArc, endLine };
+
+            return conePolyCurve;
+        }
+
+        /***************************************************/
+
+        [Description("Signed angle between two lines in XY plane.")]
+        [Input("line1", "First line to compute the angle for.")]
+        [Input("line2", "Second line to compute the angle for.")]
+        [Output("angle", "Singed angle between two lines.")]
+        private static double Angle(this Line line1, Line line2)
+        {
+            Vector line1Dir = line1.Direction();
+            Vector line2Dir = line2.Direction();
+            double angle = line1Dir.SignedAngle(line2Dir, Vector.ZAxis);
+
+            return angle;
+        }
+
+        /***************************************************/
+
+        [Description("Split line in given obstacles.")]
+        [Input("rayLine", "Line to split.")]
+        [Input("obstacles", "Polyline objects to split the line in.")]
+        [Output("splitLines", "Lines splited by the obstacles.")]
+        private static List<Line> SplitLinesInObstacles(this Line line, List<Polyline> obstacles)
+        {
+            List<Point> splitPoints = new List<Point>();
+            foreach (Polyline obstacle in obstacles)
+            {
+                foreach (Line obstLine in obstacle.SubParts())
+                {
+                    Point point = line.LineIntersection(obstLine);
+                    if (point != null)
+                        splitPoints.Add(point);
+                }
```

Comment:
Couldn't you use `LineIntersections` (written by hand, not tested)?
```suggestion
                splitPoints.AddRange(obstacle.LineIntersections(line));
```